File: app/src/main/java/net/veldor/flibustaloader/MySSLConnectionSocketFactory.java
package net.veldor.flibustaloader;


import java.io.IOException;
import java.net.InetSocketAddress;
import java.net.Proxy;
import java.net.Socket;

import javax.net.ssl.SSLContext;


import cz.msebera.android.httpclient.HttpHost;
import cz.msebera.android.httpclient.conn.ssl.SSLConnectionSocketFactory;
import cz.msebera.android.httpclient.protocol.HttpContext;


public class MySSLConnectionSocketFactory extends SSLConnectionSocketFactory {

    MySSLConnectionSocketFactory(final SSLContext sslContext) {
        super(sslContext);
    }

    @Override
    public Socket createSocket(final HttpContext context){
        InetSocketAddress socksaddr = (InetSocketAddress) context.getAttribute("socks.address");
        Proxy proxy = new Proxy(Proxy.Type.SOCKS, socksaddr);
        return new Socket(proxy);
    }

    @Override
    public Socket connectSocket(int connectTimeout, Socket socket, HttpHost host, InetSocketAddress remoteAddress,
                                InetSocketAddress localAddress, HttpContext context) throws IOException {

        InetSocketAddress unresolvedRemote = InetSocketAddress
                .createUnresolved(host.getHostName(), remoteAddress.getPort());
        return super.connectSocket(connectTimeout, socket, host, unresolvedRemote, localAddress, context);
    }
}

File: app/src/main/java/net/veldor/flibustaloader/MySSLConnectionSocketFactory.kt
package net.veldor.flibustaloader

import kotlin.Throws
import cz.msebera.android.httpclient.HttpHost
import cz.msebera.android.httpclient.conn.ssl.SSLConnectionSocketFactory
import cz.msebera.android.httpclient.protocol.HttpContext
import java.io.IOException
import java.net.InetSocketAddress
import java.net.Proxy
import java.net.Socket
import javax.net.ssl.SSLContext

class MySSLConnectionSocketFactory(sslContext: SSLContext?) :
    SSLConnectionSocketFactory(sslContext) {
    override fun createSocket(context: HttpContext): Socket {
        val socksaddr = context.getAttribute("socks.address") as InetSocketAddress
        val proxy = Proxy(Proxy.Type.SOCKS, socksaddr)
        return Socket(proxy)
    }

    @Throws(IOException::class)
    override fun connectSocket(
        connectTimeout: Int, socket: Socket, host: HttpHost, remoteAddress: InetSocketAddress,
        localAddress: InetSocketAddress, context: HttpContext
    ): Socket {
        val unresolvedRemote = InetSocketAddress
            .createUnresolved(host.hostName, remoteAddress.port)
        return super.connectSocket(
            connectTimeout,
            socket,
            host,
            unresolvedRemote,
            localAddress,
            context
        )
    }
}